src/apis/software-service.ts

```diff
@@ -0,0 +1,109 @@
+import { DocumentClient } from 'aws-sdk/clients/dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+import { SoftwareModel, Status } from './schemas/software-registry/software';
+
+const tableName = process.env.DYNAMODB_TABLE;
+
+/**
+ * Database service for software entries
+ */
+class SoftwareService {
+
+  /**
+   * Constructor
+   * @param docClient DynamoDB client
+   */
+  constructor(private readonly docClient: DocumentClient) {}
+
+  /**
+   * Creates a software entry
```

Comment:
All comments should have a line in the ts docs between the description and the first  @param, please update all